src/crewai/utilities/converter.py

```diff
@@ -72,3 +77,122 @@ def _create_chain(self):
     def is_gpt(self) -> bool:
         """Return if llm provided is of gpt from openai."""
         return isinstance(self.llm, ChatOpenAI) and self.llm.openai_api_base is None
+
+
+def convert_to_model(
+    result: str,
+    output_pydantic: Optional[Type[BaseModel]],
+    output_json: Optional[Type[BaseModel]],
+    agent: Any,
+    converter_cls: Optional[Type[Converter]] = None,
+) -> Union[dict, BaseModel, str]:
+    model = output_pydantic or output_json
+    if model is None:
+        return result
+
+    try:
+        escaped_result = json.dumps(json.loads(result, strict=False))
+        return validate_model(escaped_result, model, bool(output_json))
+    except Exception:
```

Comment:
thoughts on excepting 

```py
except json.JSONDecodeError:
        return handle_partial_json(result, model, bool(output_json), agent, converter_cls)
```

then `except Exception as e:`

excepting json.JSONDecodeError focuses on the json decode logic before throwing another instead of catching any error that can come up

Reply:
Hey @lorenzejay, so Anthropic wanted me to do this:

```python
except json.JSONDecodeError:
        return handle_partial_json(
            result, model, bool(output_json), agent, converter_cls
        )
except Exception:
        return handle_partial_json(
            result, model, bool(output_json), agent, converter_cls
        )
```

But either way, We are still calling the same thing and not reacting differently to the exceptions so I figured just do one. 

What do you think?